src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java

```diff
@@ -127,6 +127,21 @@ public OnlineStatus getOnlineStatus(@Nonnull ClientType type)
         return status == null ? OnlineStatus.OFFLINE : status;
     }
 
+    @Nonnull
+    @Override
+    public Set<ClientType> getActiveClients()
+    {
+        EnumSet<ClientType> activeClients = EnumSet.noneOf(ClientType.class);
+        this.clientStatus.forEach((client, status) ->
+        {
+            if (status != OnlineStatus.OFFLINE && status != OnlineStatus.UNKNOWN)
+            {
+                activeClients.add(client);
+            }
+        });
```

Comment:
`OFFLINE` and `UNKNOWN` should technically never be in this map, we could enforce that in the `PresenceUpdateHandler` and simplify this to `EnumSet.copyOf(clientStatus.keySet())`.

Reply:
The `OFFLINE`/`UNKNOWN` check seems to be already handled in `Member#setOnlineStatus(...)`:

https://github.com/DV8FromTheWorld/JDA/blob/feature/member_active_clients/src/main/java/net/dv8tion/jda/internal/entities/MemberImpl.java#L292-L301